app/javascript/packs/cms.ts

```diff
@@ -332,14 +312,25 @@ function setUpDropdownButtonClose () {
 }
 
 function closeAllDropdowns (exception?: HTMLElement) {
-  document.querySelectorAll('.dropdown.expanded')
+  document.querySelectorAll('.pf-c-dropdown.pf-m-expanded')
     .forEach(dropdown => {
-      const buttonGroup = dropdown.parentElement as HTMLDivElement
-
-      if (exception && buttonGroup.contains(exception)) {
+      if (exception && dropdown.contains(exception)) {
         return
       }
-
-      dropdown.classList.remove('expanded')
+      dropdown.classList.remove('pf-m-expanded')
+      dropdown.querySelector("ul")!.style.display = 'none'
     })
 }
+
+/**
+ * Set up CMS Preview button to act as a link
+ */
+function setUpPreviewButton () {
+  $(document).on('click', '#cms-preview-button button.pf-c-dropdown__toggle-button:not(.dropdown-toggle)', (event) => {
+    const url = event.target.dataset.url
+    if (url) {
```

Comment:
Is it ever going to be undefined?

Reply:
Hmm... in theory, `data-url` should always be present, but I don't know... if for some reason (a developer error etc.) it's not there, I think it's not a bad idea to check.